cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -121,7 +121,7 @@ struct ZonedLocalizer {
 
 //
 // Executor class for temporal component extractors, i.e. scalar kernels
-// with the signature Timestamp -> <non-temporal scalar type `OutType`>
+// with the signature Timestamp|Date32|Date64 -> <non-temporal scalar type `OutType`>
```

Comment:
`temporal types` you use somewhere else sounds better than `Timestamp|Date32|Date64` IMO.